pkg/server/server.go

```diff
@@ -172,13 +179,15 @@ func (s *ProxyServer) getFrontend(agentID string, connID int64) (*ProxyClientCon
 
 // NewProxyServer creates a new ProxyServer instance
 func NewProxyServer(serverID string, serverCount int, agentAuthenticationOptions *AgentTokenAuthenticationOptions) *ProxyServer {
+	bm := NewDefaultBackendManager()
 	p := &ProxyServer{
 		frontends:                  make(map[string](map[int64]*ProxyClientConnection)),
 		PendingDial:                NewPendingDialManager(),
 		serverID:                   serverID,
 		serverCount:                serverCount,
-		BackendManager:             NewDefaultBackendManager(),
+		BackendManager:             bm,
 		AgentAuthenticationOptions: agentAuthenticationOptions,
+		Readiness:                  bm,
 	}
 	return p
```

Comment:
Maybe simply return without the extra variable? 
```go 
	return  &ProxyServer{
		frontends:                  make(map[string](map[int64]*ProxyClientConnection)),
		PendingDial:                NewPendingDialManager(),
		serverID:                   serverID,
		serverCount:                serverCount,
		BackendManager:             NewDefaultBackendManager(),
		BackendManager:             bm,
		AgentAuthenticationOptions: agentAuthenticationOptions,
		Readiness:                  bm,
	}
```

Reply:
Done.